src/utils/layout/hierarchy.ts

```diff
@@ -120,6 +122,8 @@ export function dataSourcesFromState(state: IRuntimeState): HierarchyDataSources
   };
 }
 
+export const selectDataSourcesFromState = createSelector(dataSourcesFromState, (data) => data);
```

Comment:
Memoizing this selector since it returns an object. This means it will return a different result each time the selector runs, and creates a large warning in the console.